controllers/imagecollector/imagecollector_controller.go

```diff
@@ -154,7 +155,13 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	go func() {
+	delay := *repeatPeriod
+	if *repeatImmediate {
```

Comment:
what is this for? is this different than setting repeat period to 0?

Reply:
The collect-scan-erase process can either begin immediately or after the specified delay. Our default has been to start immediately, with no option to change that behavior. This change makes it possible to do either.

The only reason I added this to this PR is because the collector pods were messing up the tests. Some tests (for example `imagelist_rm_images`) test the behavior of the imagelist, and have to jump through hoops to not conflict with the collector which begins immediately after deployment. example: https://github.com/Azure/eraser/blob/76aa58d70fb1cce6f2c61f466dd4b34dceb470b5/test/e2e/tests/imagelist_rm_images/eraser_test.go#L51

Ideally I'd like to move to a more robust scheduling system (example: allow the user to specify a cron string).

Reply:
> what is this for? is this different than setting repeat period to 0?

Sorry, misunderstood your question. Yes, it's different. If we set the repeat period to zero it would start again immediately. Here's how it works:
- If repeatPeriod = 30 min and repeatImmediate = true, then we will run one job at startup, then the next job 30 minutes after that
- If repeatPeriod = 30 min and repeatImmediate = false, then we will wait 30 mins after startup to run the first job

Reply:
I see, `repeat` seems a bit misleading. perhaps we can name it something like `schedule-immediate` (or anything else you can think of)?